devices/uart-16550/uarthw-riscv.c

```diff
@@ -0,0 +1,68 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Operating system loader
+ *
+ * UART 16550 (RISC-V)
+ *
+ * Copyright 2020, 2021 Phoenix Systems
+ * Author: Julia Kosowska, Pawel Pisarczyk, Lukasz Kosinski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include <hal/hal.h>
+#include <lib/errno.h>
+
+#include "uart-16550.h"
+
+
+typedef struct {
+	void *base;
+	unsigned int irq;
+} uarthw_context_t;
+
+
+static const uarthw_context_t uarts[] = {
+	{ (void *)0x10000000, 10 }
+};
+
+
+unsigned char uarthw_read(void *hwctx, unsigned int reg)
+{
+	unsigned int addr = (unsigned int)((uarthw_context_t *)hwctx)->base;
+
+	return *(volatile unsigned char *)(addr + reg);
+}
+
+
+void uarthw_write(void *hwctx, unsigned int reg, unsigned char val)
+{
+	unsigned int addr = (unsigned int)((uarthw_context_t *)hwctx)->base;
+
+	*(volatile unsigned char *)(addr + reg) = val;
```

Comment:
This can be done on one line, no need to cast to volatile (if you make base volatile)

Reply:
Fixed.